### ISSUE TITLE
How enable private image scan on ECR?

### ISSUE THREAD

**Issue by jmtt89** [castai/kvisor#193]
Hi, 

the documentation shows an example to use username/password to enable private repositories, but on ECR the way to obtain the password is using `ecr get-login-password`

i see on code exist a variable to support ECR 

```go
type DockerOption struct {
	// Auth
	UserName string `yaml:"user_name"`
	Password string `yaml:"password"`

	// RegistryToken is a bearer token to be sent to a registry
	RegistryToken string `yaml:"registry_token"`

	// ECR
	AwsAccessKey    string `yaml:"aws_access_key"`
	AwsSecretKey    string `yaml:"aws_secret_key"`
	AwsSessionToken string `yaml:"aws_session_token"`
	AwsRegion       string `yaml:"aws_region"`

	// GCP
	GcpCredPath string `yaml:"gcp_cred_path"`

	// SSL/TLS
	InsecureSkipTLSVerify bool `yaml:"insecure_skip_tls_verify"`
	NonSSL                bool `yaml:"non_ssl"`
}
```

but i don't find any example of how to use it.

**Comment by anjmao** [comment]
Hi @jmtt89 

Will double check and test ECR setup. Thanks for reporting!

**Comment by domust** [comment]
Hi, there are a couple of options to enable private image scan:
1. If you already have an image pull secret, you could just copy it over to the namespace where kvisor is installed and configure kvisor to use it via:
```sh
helm upgrade castai-kvisor castai-helm/castai-kvisor -n castai-agent \
  --reuse-values --set castai.imageScanSecret=[secret-name]
```
2. If you don't have an image pull secret, [ECR documentation](https://docs.aws.amazon.com/AmazonECR/latest/userguide/registry_auth.html#registry-auth-token) explains how to authenticate to docker, which requires exactly the same data as an image pull secret, for example:
```sh
kubectl -n castai-agent create secret docker-registry [secret-name] \
  --docker-server=${AWS_ACCOUNT}.dkr.ecr.${AWS_REGION}.amazonaws.com \
  --docker=username=AWS \
  --docker-password=$(aws ecr get-login-password)
```
I have also updated [documentation](https://docs.cast.ai/docs/kvisor#enable-private-image-scan-v0370) to include an example.
3. If you don't want to share your image pull secrets, there's also an option to enable autoscaler which allows kvisor to access images directly on the worker nodes after kubelet downloads them.

I hope this helps! I'm closing the issue. If this doesn't help, please reopen it and let us know what's missing/not working.

**Comment by jmtt89** [comment]
Hi @domust thanks for the answer.

Unfortunately, the generated password/token from the [get-login-password](https://docs.aws.amazon.com/cli/latest/reference/ecr/get-login-password.html) command is only valid for 12 hours, so the solution provided require update kube secret every 12 hours (maybe with cronjob)

**Comment by anjmao** [comment]
@jmtt89 Do you use EKS. It is possible to create IAM in AWS add configure kvisor Kubernetes service account to access ECR without any password or credentials. This is the most secure and recommended way. We will look into this workload and test if everything is supported.

**Comment by anjmao** [comment]
Tested ecr managed registry. After some fixes https://github.com/castai/kvisor/pull/194 private images can be scanned without any secret passwords. Will merge it tomorrow and update docs how to enable it.

**Comment by jmtt89** [comment]
Ohhh perfect, thanks @anjmao

**Comment by anjmao** [comment]
@jmtt89 Hi, I added docs here https://docs.cast.ai/docs/images-scanning#amazon-elastic-container-registry-amazon-ecr

It should be available in latest v0.50.0 release. Let me know if it works for you. I was able to scan private images from my ECR.

**Comment by jmtt89** [comment]
Thanks @anjmao, I tested it, and it works correctly, the only detail that may be missing is if using helm chart needs set imageScanServiceAccount.create to false.
